crawler/file_processing.py

```diff
@@ -1138,48 +1157,48 @@ def row_channel_result_values_valid(self, row, line_number) -> bool:
 
         return True
 
-    def is_within_cq_range(self, range_min, range_max, num) -> bool:
+    @staticmethod
+    def is_within_cq_range(range_min: Decimal, range_max: Decimal, num: ModifiedRowValue) -> bool:
         """Validation to check if a number lies within the expected range.
 
         Arguments:
             range_min {Decimal} - minimum range number
             range_max {Decimal} - maximum range number
-            num {Decimal128} - the number to be tested
+            num {ModifiedRowValue} - the number to be tested
 
         Returns:
             bool - whether the value lies within range
         """
-        # cannot compare Decimal128 to Decimal or to other Decimal128s
-        return range_min <= num.to_decimal() <= range_max
+        # TODO: this casting is dangerous and needs to be re-thought...
+        return range_min <= Decimal(cast(Decimal128, num).to_decimal()) <= range_max
 
-    def is_row_channel_cq_in_range(self, row, line_number, fieldname) -> bool:
+    def is_row_channel_cq_in_range(self, row: ModifiedRow, line_number: int, fieldname: str) -> bool:
         """Is the channel cq within the specified range.
 
         Arguments:
-            row {Row} - row object from the csvreader
-            line_number {integer} - line number within the file
+            row {ModifiedRow} - row object from the CSV DictReader
+            line_number {int} - line number within the file
             fieldname {str} - the name of the cq column
 
         Returns:
             bool - whether the cq value is valid
         """
-        if row.get(fieldname):
-            if not self.is_within_cq_range(MIN_CQ_VALUE, MAX_CQ_VALUE, row.get(fieldname)):
-                self.logging_collection.add_error(
-                    "TYPE 20",
-                    f"{fieldname} not in range ({MIN_CQ_VALUE}, {MAX_CQ_VALUE}), "
-                    f"line: {line_number}, result: {row.get(fieldname)}",
-                )
-                return False
+        if fieldname in row and not self.is_within_cq_range(MIN_CQ_VALUE, MAX_CQ_VALUE, row[fieldname]):
```

Comment:
This pattern was changed previously - see this commit - https://github.com/sanger/crawler/commit/fcc23fc07a6448d9cf2d7206966044197c2832cf 

I am trying to remember what the difference is but I think we thought quite hard about it before.

Maybe `row.get(fieldname)` checks if `fieldname in row` and `row[fieldname] is not None` or something?

If it turns out that `row.get(fieldname)` turns out to be correct, probably worth putting in some tests to avoid this back and forth in future?

Reply:
Oh OK, thanks and good remembering! Will have a look a little deeper here too and add comments to prevent back and forth too.

Reply:
So it was my initial misunderstanding of the `.get()` method. I think it is more readable this way (checking for the key in the dict first) but using the `.get()` method is more readable on line 1082 - soon to be updated and pushed. What do you think?

Reply:
Yes, `row.get(fieldname)` returns the value of the dict with key fieldname (or None if there is no such key) but that value could be a boolean which is misleading. I think it's more readable to first check if that key exists and then do the necessary checks on the logic?